realis.go

```diff
@@ -268,34 +402,72 @@ func (r *realisClient) ReestablishConn() error {
 	fmt.Println("ReestablishConn begin ....")
 	r.Close()
 	//First check cluster object for re-establish; if not available then try with scheduler url.
+	//var config *RealisConfig
+	var err error
+	var url string
+
 	if r.config.cluster != nil && r.config.username != "" && r.config.password != "" {
 		//Re-establish using cluster object.
-		url, err := LeaderFromZK(*r.config.cluster)
+		url, err = LeaderFromZK(*r.config.cluster)
 		if err != nil {
 			fmt.Errorf("LeaderFromZK error: %+v\n ", err)
 		}
 		fmt.Println("ReestablishConn url: ", url)
-		config, err := newDefaultConfig(url, 10000)
+		if r.config.jsonTransport {
```

Comment:
Just an idea, but couldn't we store the function that create the transport (i.e newTBinaryConfig or newTJSONConfig) and call them instead of duplicating code? (Seems to work on a small test but maybe this might become more complicated https://play.golang.org/p/TPkH93kI_F)

Something like
r.protoConfig = newTJSONConfig or r.protoConfig = newTBinaryConfig

Then when reconnecting:
r.protoConfig(url, r.config.timeoutms)